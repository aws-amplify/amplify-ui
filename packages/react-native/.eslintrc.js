module.exports = {
  ignorePatterns: ['dist', '.eslintrc.js', '*.config.js', 'jest.setup.js'],
  extends: ['@aws-amplify/amplify-ui/react'],
  overrides: [
    {
      extends: ['@aws-amplify/amplify-ui/jest'],
      files: ['**/__mocks__/**', '**/__tests__/**'],
    },
  ],
  // point to local tsconfig
  parserOptions: { project: ['tsconfig.json'], tsconfigRootDir: __dirname },
};
