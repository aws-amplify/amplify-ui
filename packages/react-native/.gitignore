dist
.watchmanconfig

# ignore jest generated coverage directory
coverage 
