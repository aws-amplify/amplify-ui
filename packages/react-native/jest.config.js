module.exports = {
  preset: 'react-native',
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  collectCoverage: true,
  collectCoverageFrom: [
    '<rootDir>/src/**/*.{js,jsx,ts,tsx}',
    '!<rootDir>/src/**/*{c,C}onstants.ts',
  ],
  moduleNameMapper: {
    '^react$': '<rootDir>/node_modules/react',
    '^react-native$': '<rootDir>/node_modules/react-native',
  },
  coverageThreshold: {
    global: {
      branches: 90,
      functions: 90,
      lines: 90,
      statements: 90,
    },
  },
  setupFiles: ['<rootDir>/jest.setup.js'],
};
