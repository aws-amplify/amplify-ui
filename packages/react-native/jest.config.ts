import { Config } from 'jest';

const config: Config = {
  preset: 'react-native',
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  collectCoverage: true,
  collectCoverageFrom: [
    '<rootDir>/src/**/*.{js,jsx,ts,tsx}',
    '!<rootDir>/src/**/*{c,C}onstants.ts',
  ],
  moduleNameMapper: {
    '^react-native$': '<rootDir>/node_modules/react-native',
    '^uuid$': '<rootDir>/../../node_modules/uuid',
  },
  modulePaths: ['<rootDir>/node_modules/'],
  coverageThreshold: {
    global: {
      branches: 90,
      functions: 90,
      lines: 90,
      statements: 90,
    },
  },
  setupFiles: ['<rootDir>/jest.setup.ts'],
};

export default config;
