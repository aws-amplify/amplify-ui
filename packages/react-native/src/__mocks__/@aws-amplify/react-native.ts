export default {
  loadUrlPolyfill: jest.fn(),
};
