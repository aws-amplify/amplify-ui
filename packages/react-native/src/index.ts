export {
  Authenticator,
  AuthenticatorProps,
  useAuthenticator,
  UseAuthenticator,
  withAuthenticator,
  WithAuthenticatorOptions,
} from './Authenticator';
export * from './InAppMessaging';
export {
  Theme,
  ThemeProvider,
  defaultDarkModeOverride,
  defaultTheme,
  useTheme,
} from './theme';
