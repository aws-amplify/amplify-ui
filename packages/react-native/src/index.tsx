import React from 'react';
import { Text, View } from 'react-native';

export function HelloWorld(): JSX.Element {
  return (
    <View>
      <Text>Hello World!</Text>
    </View>
  );
}
