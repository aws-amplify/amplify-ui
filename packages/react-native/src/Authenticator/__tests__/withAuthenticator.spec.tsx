import React from 'react';
import { Text } from 'react-native';
import { render } from '@testing-library/react-native';

import * as UIReactCoreModule from '@aws-amplify/ui-react-core';
import {
  AuthenticatorMachineContext,
  UseAuthenticator,
} from '@aws-amplify/ui-react-core';

import { withAuthenticator } from '..';

const CHILD_TEST_ID = 'child-test-id';
const CHILD_CONTENT = 'Test Children';

function TestChildren() {
  return <Text testID={CHILD_TEST_ID}>{CHILD_CONTENT}</Text>;
}
const TestComponent = () => {
  return null;
};

const TestApp = withAuthenticator(TestChildren);

const useAuthenticatorSpy = jest.spyOn(UIReactCoreModule, 'useAuthenticator');
const useAuthenticatorRouteSpy = jest.spyOn(
  UIReactCoreModule,
  'useAuthenticatorRoute'
);

type MockSelectorParam =
  | ((
      context: AuthenticatorMachineContext
    ) => AuthenticatorMachineContext[keyof AuthenticatorMachineContext][])
  | undefined;

describe('withAuthenticator', () => {
  beforeEach(() => {
    jest.clearAllMocks();

    useAuthenticatorRouteSpy.mockReturnValue({
      Component: TestComponent as any,
      props: {} as any,
    });
  });

  it('renders as expected', () => {
    const mockUseAuthenticator = (cb: MockSelectorParam) =>
      cb?.({
        route: 'signIn',
      } as unknown as AuthenticatorMachineContext)[0] as unknown as UseAuthenticator;
    useAuthenticatorSpy.mockImplementation(mockUseAuthenticator);

    const { toJSON, queryByTestId } = render(<TestApp />);

    expect(queryByTestId(CHILD_TEST_ID)).toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders children when route is authenticated', () => {
    useAuthenticatorSpy.mockImplementation(
      () =>
        ({
          authStatus: 'authenticated',
        } as unknown as UIReactCoreModule.UseAuthenticator)
    );

    const { toJSON, getByTestId } = render(<TestApp />);

    expect(getByTestId(CHILD_TEST_ID)).toBeDefined();
    expect(toJSON()).toMatchSnapshot();
  });
});
