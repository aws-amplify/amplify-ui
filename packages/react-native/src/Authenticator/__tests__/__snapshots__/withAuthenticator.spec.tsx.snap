// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAuthenticator renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 12,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`withAuthenticator renders children when route is authenticated 1`] = `
<Text
  testID="child-test-id"
>
  Test Children
</Text>
`;
